narwhals/_spark_like/dataframe.py

```diff
@@ -105,8 +110,38 @@ def select(
 
             return self._from_native_frame(spark_df)
 
-        new_columns_list = [col.alias(col_name) for col_name, col in new_columns.items()]
-        return self._from_native_frame(self._native_frame.select(*new_columns_list))
+        if all(returns_scalar):
+            new_columns_list = [
+                col.alias(col_name) for col_name, col in new_columns.items()
+            ]
+            return self._from_native_frame(self._native_frame.agg(*new_columns_list))
+        else:
+            new_columns_list = [
+                col.over(Window.partitionBy(F.lit(1))).alias(col_name)
+                if _returns_scalar
+                else col.alias(col_name)
+                for (col_name, col), _returns_scalar in zip(
+                    new_columns.items(), returns_scalar
+                )
+            ]
```

Comment:
i think this is may be too late to set `over`

for example, in `nw.col('a') - nw.col('a').mean()`  - I think it's in the binary operation `__sub__` that `nw.col('a').mean()` needs to become `nw.col('a').mean().over(lit(1))`

as in, we want to translate `nw.col('a') - nw.col('a').mean()` to `F.col('a') - F.col('a').mean().over(F.lit(1))`. the code, however, as far as I can tell, translates it to `(F.col('a') - F.col('a').mean()).over(F.lit(1))`

Reply:
That happens in `maybe_evaluate`, and you can see that now the `reduction_test` are passing.
I know it's not ideal to have it in two places, but I couldn't figure out a unique place in which to handle this as `maybe_evaluate` is called only to evaluate _other_ arguments

Reply:
ah I see! yes this might be fine then, thanks!